keras2onnx/topology.py

```diff
@@ -233,27 +233,34 @@ def convert_topology(topology, model_name, doc_string, target_opset, channel_fir
         keras2onnx_logger().debug("Converting the operator (%s): %s" % (operator.full_name, operator.type))
         get_converter(operator.type)(scope, operator, container)
 
-    # When calling ModelComponentContainer's add_initializer(...), nothing is added into the input list. However, in
-    # ONNX initializers should also be model's (GraphProto) inputs. Thus, we create ValueInfoProto objects from
-    # initializers (type: TensorProto) directly and then add them into model's input list.
+    # When calling ModelComponentContainer's add_initializer(...), nothing is added into the input list.
+    # However, In ONNX, for target opset < 9, initializers should also be model's (GraphProto) inputs.
+    # Thus, we create ValueInfoProto objects from initializers (type: TensorProto) directly and then add them into model's input list.
     extra_inputs = []  # ValueInfoProto list of the initializers
-    for tensor in container.initializers:
-        # Sometimes (especially when creating optional input values such as RNN's initial hidden state), an initializer
-        # is also one of the original model's input, so it has been added into the container's input list. If this is
-        # the case, we need to skip one iteration to avoid duplicated inputs.
-        if tensor.name in [value_info.name for value_info in container.inputs]:
-            continue
-
-        # Initializers are always tensors so we can just call make_tensor_value_info(...)
-        value_info = helper.make_tensor_value_info(tensor.name, tensor.data_type, tensor.dims)
-        extra_inputs.append(value_info)
-
+    if target_opset < 9:
+        for tensor in container.initializers:
+            # Sometimes (especially when creating optional input values such as RNN's initial hidden state), an initializer
+            # is also one of the original model's input, so it has been added into the container's input list. If this is
+            # the case, we need to skip one iteration to avoid duplicated inputs.
+            if tensor.name in [value_info.name for value_info in container.inputs]:
+                continue
+
+            # Initializers are always tensors so we can just call make_tensor_value_info(...)
+            value_info = helper.make_tensor_value_info(tensor.name, tensor.data_type, tensor.dims)
+            extra_inputs.append(value_info)
 
     # enable the ONNX optimizations
     try:
         import onnxtk
-        nodes = onnxtk.optimizer.optimize_onnx(container.nodes, nchw_inputs=nchw_inputs, inputs=container.inputs + extra_inputs,
-                              outputs=container.outputs)
+        if target_opset < 9:
+            nodes = onnxtk.optimizer.optimize_onnx(container.nodes, nchw_inputs=nchw_inputs,
```

Comment:
where is onnxtk change?

Reply:
https://github.com/onnx/onnxmltools/pull/263